okio/src/test/java/okio/ReadUtf8LineTest.java

```diff
@@ -94,6 +94,20 @@
     }
   }
 
+  @Test public void readUtf8LineStrictWithLimits() throws IOException {
+    data.writeUtf8("ab\nde\n");
+    assertEquals("ab", source.readUtf8LineStrict(4));
+    assertEquals("de", source.readUtf8LineStrict(4));
+
+    data.writeUtf8("abcdef\n");
+    try {
+      source.readUtf8LineStrict(4);
+      fail();
+    } catch (EOFException expected) {
+      assertEquals("\\n not found: size=4 content=6162636465660a…", expected.getMessage());
```

Comment:
Rather than including encoded bytes in the exception message it'd be nice to have a guarantee (JavaDoc and tests) that the BufferedSource is still in a useable state so users could fetch the information themselves if necessary.